crates/requirements-txt/src/lib.rs

```diff
@@ -317,6 +317,27 @@ pub struct RequirementsTxt {
     pub no_index: bool,
 }
 
+/// Calculates column and line based on the cursor and content.
+fn calculate_line_column_pair(content: &str, position: usize) -> (usize, usize) {
+    let mut line = 1;
+    let mut column = 1;
+
+    for (index, char) in content.char_indices() {
+        if index >= position {
+            break;
+        }
+        // This should work fine for both Windows and Linux line endings
+        if char == '\n' {
+            line += 1;
+            column = 1;
+        } else if char != '\r' {
+            column += 1;
```

Comment:
I would suggest using the [`unicode-width`](https://docs.rs/unicode-width/latest/unicode_width/) crate to compute the visual width of the codepoint via [`UnicodeWidthChar::width`](https://docs.rs/unicode-width/latest/unicode_width/trait.UnicodeWidthChar.html#tymethod.width).

Reply:
I don't think we should be using `width` here. I would need to check again but what I remember is that editors use character offsets for column indices and not their width. Ruff also uses character offsets and not the width for diagnostics.

Reply:
Yeah I agree that we should do whatever editors are likely to use here.

Reply:
Yeah, I think char_indices is okay here, but I would like to tweak the newline handling slightly in line with the Ruff reference above.